lib/tpm2_kdfa.c

```diff
@@ -21,8 +18,7 @@ TSS2_RC tpm2_kdfa(TPMI_ALG_HASH hash_alg, TPM2B *key, char *label,
     tpm2b_i_2.size = 4;
 
     tpm2b_bits.size = 4;
-    bits_swizzled = tpm2_util_endian_swap_32(bits);
-    *(UINT32 *) tpm2b_bits_ptr = bits_swizzled;
+    *(UINT32 *)&tpm2b_bits.buffer[0] = tpm2_util_hton_32(bits);
```

Comment:
```suggestion
    UINT8 *tpm2b_buffer_ptr = &tpm2b_bits.buffer[0];
    *(UINT32 *) tpm2b_buffer_ptr  = tpm2_util_hton_32(bits);
```

Reply:
Your suggestion doesn't work for me (gcc 7.5.0 on s390x):
```
lib/tpm2_kdfa.c: In function ‘tpm2_kdfa’:
lib/tpm2_kdfa.c:22:5: error: invalid operands to binary * (have ‘BYTE * {aka unsigned char *}’ and ‘UINT32 * {aka unsigned int *}’)
     UINT8 *tpm2b_buffer_ptr = &tpm2b_bits.buffer[0]
                               ~~~~~~~~~~~~~~~~~~~~~
     *(UINT32 *) tpm2b_buffer_ptr = tpm2_util_hton_32(bits);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Pushed another solution now. Could also fall back to the previous code...

Reply:
... with the `tpm2_util_hton_32()` applied of course

Reply:
> Pushed another solution now. Could also fall back to the previous/original code...

I will post feedback with the build status shortly.